lib/keila_web/controllers/public_form_controller.ex

```diff
@@ -126,43 +126,96 @@ defmodule KeilaWeb.PublicFormController do
 
   @default_unsubscribe_form %Contacts.Form{settings: %Contacts.Form.Settings{}}
   @spec unsubscribe(Plug.Conn.t(), map()) :: Plug.Conn.t()
-  def unsubscribe(conn, %{
+  def unsubscribe(conn = %{method: "GET"}, %{
         "project_id" => project_id,
         "recipient_id" => recipient_id,
         "hmac" => hmac
       }) do
     if Mailings.valid_unsubscribe_hmac?(project_id, recipient_id, hmac) do
+      form = Contacts.get_project_forms(project_id) |> List.first() || @default_unsubscribe_form
+
+      conn
+      |> put_meta(:title, gettext("Unsubscribe"))
+      |> assign(:form, form)
+      |> assign(:project_id, project_id)
+      |> assign(:recipient_id, recipient_id)
+      |> assign(:hmac, hmac)
+      |> assign(:mode, :full)
+      |> assign(:unsubscribed, false)
+      |> render("unsubscribe.html")
+    else
+      conn |> put_status(404) |> halt()
+    end
+  end
+
+  def unsubscribe(conn = %{method: "POST"}, %{
+        "project_id" => project_id,
+        "recipient_id" => recipient_id,
+        "hmac" => hmac
+      }) do
+    # Validate HMAC and confirm parameter for security
+    if Mailings.valid_unsubscribe_hmac?(project_id, recipient_id, hmac) and
+       conn.params["confirm_unsubscribe"] == "true" do
```

Comment:
This would need to match the List-Unsubscribe-Post header. Also it should probably just be called "confirm".